pkg/gatewayserver/scheduling/scheduler.go

```diff
@@ -66,19 +67,32 @@ type RTTs interface {
 
 // NewScheduler instantiates a new Scheduler for the given frequency plan.
 // If no time source is specified, the system time is used.
-func NewScheduler(ctx context.Context, fp *frequencyplans.FrequencyPlan, enforceDutyCycle bool, timeSource TimeSource) (*Scheduler, error) {
+func NewScheduler(ctx context.Context, fp *frequencyplans.FrequencyPlan, enforceDutyCycle bool, scheduleAnytimeDelay *time.Duration, timeSource TimeSource) (*Scheduler, error) {
+	logger := log.FromContext(ctx)
 	if timeSource == nil {
 		timeSource = SystemTimeSource
 	}
+
+	if scheduleAnytimeDelay == nil || *scheduleAnytimeDelay == 0 {
+		scheduleAnytimeDelay = &ScheduleTimeLong
+	} else if *scheduleAnytimeDelay < ScheduleTimeShort {
+		logger.WithFields(log.Fields(
+			"schedule_anytime_delay_minimum", ScheduleTimeShort.String(),
+			"schedule_anytime_delay_requested", scheduleAnytimeDelay.String(),
```

Comment:
- Context is clear; you can use `minimum` and `requested`
- Let the logger handle the stringing, but you can dereference (`*scheduleAnytimeDelay`)